contracts/message-bus/MessageInbox.sol

```diff
@@ -121,24 +119,16 @@ contract MessageInbox is MessageBox, Proof {
             "State root provider address is 0."
         );
 
-        require(
-            _verifyingAddress != address(0),
-            "Verifying address is 0."
-        );
-
         messageOutbox = _messageOutbox;
         outboxStorageIndex = _outboxStorageIndex;
 
-        inboundMessageIdentifier = keccak256(
-            abi.encode(
-                DOMAIN_SEPARATOR_TYPEHASH,
-                DOMAIN_SEPARATOR_NAME,
-                DOMAIN_SEPARATOR_VERSION,
-                _metachainId,
-                _verifyingAddress
-            )
+        inboundChannelIdentifier = MessageBox.hashChannelIdentifier(
```

Comment:
According to the ticket, calculation of the inbound channel identifier should be done in the following order (param order):

```solidity
bytes32 inboundChannelIdentifier = hashChannelIdentifier(
    metachainId,
    inboxAddress,
    outboxAddress
);
```

Reply:
Thank you @pgev for the comment.
Checked with @deepesh-kn and above is little confusing 😕and not correctly explained in the ticket.
Below is  hashChannelIdentifier interface: 
```
function hashChannelIdentifier(
        bytes32 _metachainId,
        address _outbox,
        address _inbox
    )
```
For ConsensusGateway => inboundChannelIdentifier
- outboxAddress is consensusCoGateway
- inbox address is ConsensusGateway (address(this) is ConsensusGateway address here after inheritance)